rest/api.go

```diff
@@ -406,10 +406,11 @@ type DatabaseRoot struct {
 	RequireResync                 []string `json:"require_resync,omitempty"`
 }
 
-type dbSummary struct {
-	DBName string `json:"db_name"`
-	Bucket string `json:"bucket"`
-	State  string `json:"state"`
+type DbSummary struct {
+	DBName        string `json:"db_name"`
+	Bucket        string `json:"bucket"`
+	State         string `json:"state"`
+	OfflineReason string `json:"offline_reason,omitempty"`
```

Comment:
We're only in one state at a time, so I think a simple 'reason' property is preferable.  (e.g. in future if we want to provide a reason for state=Resyncing, we can reuse).  What do you think?

Reply:
I think I'd prefer to do create a parameter that is `RequireResync` to match the /db/ endpoint, and there isn't ambiguity.

I think the status field offers a lot of flexibility, but it comes at some downsides, which is that it is not always populated. The reason that `reason` it might not be populated is opaque, so if the user calls `/db/_offline` or   sets `start_offline: true` should the reason be offline in that case?

Capella doesn't use the /db/ `require_resync` field, and implementing this take the place of that field.

Reply:
Since require_resync can potentially be very large, I don't think it's necessarily appropriate for a list REST endpoint.  I think it makes sense to keep the _all_dbs response relatively lightweight (to support frequent polling), and keep `require_resync` available on the /db/ endpoint.  That gives consumers the option to fetch the data based on the reason in the list view, but not be committed to it on every polling interval.